pkg/plugin/state.go

```diff
@@ -569,98 +569,205 @@ func (e *AutoscaleEnforcer) handleNodeDeletion(logger *zap.Logger, nodeName stri
 	logger.Info("Deleted node")
 }
 
-func (e *AutoscaleEnforcer) handlePodStarted(logger *zap.Logger, pod *corev1.Pod) {
-	podName := util.GetNamespacedName(pod)
+func (e *AutoscaleEnforcer) handleStarted(logger *zap.Logger, pod *corev1.Pod) {
 	nodeName := pod.Spec.NodeName
 
 	logger = logger.With(
 		zap.String("action", "Pod started"),
-		zap.Object("pod", podName),
 		zap.String("node", nodeName),
+		util.PodNameFields(pod),
 	)
-
-	if pod.Spec.SchedulerName == e.state.conf.SchedulerName {
-		logger.Info("Got non-VM pod start event for pod assigned to this scheduler; nothing to do")
-		return
+	if migrationName := util.TryPodOwnerVirtualMachineMigration(pod); migrationName != nil {
+		logger = logger.With(zap.Object("virtualmachinemigration", *migrationName))
 	}
 
-	logger.Info("Handling non-VM pod start event")
+	logger.Info("Handling Pod start event")
+
+	_, _, _ = e.reserveResources(context.TODO(), logger, pod, "Pod started", false)
+}
+
+func (e *AutoscaleEnforcer) reserveResources(
+	ctx context.Context,
+	logger *zap.Logger,
+	pod *corev1.Pod,
+	action string,
+	allowDeny bool,
+) (ok bool, _ *verdictSet, _ error) {
```

Comment:
Why we return both an error and a boolean here?

Reply:
Elaborated in a comment in f4b9d26cd4a19be1baec6fdf0899f04c79d2b9ac — wdyt?